lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,86 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::core::ILogger &log, io::I2c &i2c, const std::uint8_t channel)
+    : logger_(log),
+      i2c_(i2c),
+      channel_(channel)
+{
+}
+
+Temperature::~Temperature()
+{
+}
+
+std::optional<std::int16_t> Temperature::read()
+{
+  const auto status_check_result = i2c_.readByte(kTemperatureDefaultAddress, kStatus);
+  if (!status_check_result) {
+    logger_.log(hyped::core::LogLevel::kFatal,
+                "Failed to read temperature sensor status at channel %d",
+                channel_);
+    return std::nullopt;
+  }
+  if (status_check_result.value() == kBusy) {
```

Comment:
Could this not be a switch instead of a series of if statements to check `status_check_result.value()`? And if they are if statements checking one value there should be else-if statements